apps/lockfile-explorer-web/src/parsing/compareSpec.ts

```diff
@@ -0,0 +1,57 @@
+import { IPackageJson } from '../types/IPackageJson';
+
+export interface ISpecChange {
+  type: string;
+  pkg: string;
+  from?: string;
+  to?: string;
+}
+
+export const compareSpec = (
+  packageJson: IPackageJson,
+  packageSpec: IPackageJson
+): Map<string, ISpecChange> => {
+  const pkgJsonMap = new Map();
+  const pkgSpecMap = new Map();
+  for (const [entry, version] of Object.entries({
```

Comment:
The key/value types are missing, e.g. `Map<string, string>()`.

Wherever a `Map` is declared, the code should clearly indicate what they keys and values are, because this is crucial information for understanding the data structure.

The bare minimum would be a code comment like this:

```ts
// ISBN --> title of the book
const books: Map<string, string> = new Map();
```

My personal preference is to use `values-by-key` naming pattern with an example:
```ts
// Example: "978-1400079988" --> "War and Peace"
const bookTitleByIsbn: Map<string, string> = new Map();
``